scripts/release/build/image_build_process.py

```diff
@@ -109,6 +117,25 @@ def check_if_image_exists(self, image_tag: str) -> bool:
         else:
             return True
 
+    def pull_image(self, image):
+        docker_cmd = python_on_whales.docker
+        try:
+            docker_cmd.image.pull(image, quiet=True)
+        except Exception as e:
+            logger.error(f"Failed pulling image. {e}")
+            raise e
+
+    def get_manfiest_list_digest(self, image) -> Optional[str]:
+        self.pull_image(image)
+
+        docker_cmd = python_on_whales.docker
+        try:
+            manifest = docker_cmd.image.inspect(image)
+            return manifest.id
+        except Exception as e:
+            logger.error(f"Failed inspecting the image to get manifest list digest. {e}")
+            raise e
```

Comment:
```suggestion
        except Exception as e:
            raise Exception("Failed inspecting the image to get manifest list digest") from e
```

This is idiomatic way to chain exceptions in Python -> https://docs.python.org/3/tutorial/errors.html#exception-chaining. I also see you logging the error immediately when caught, but this is not needed if we re-throw it or chain with new exception. The exception will be logged by the program either way.